src/aind_metadata_mapper/neuropixels/rig.py

```diff
@@ -0,0 +1,197 @@
+"""ETL class for neuropixels rigs."""
+
+import json
+import yaml
+import pathlib
+import datetime
+import configparser
+from aind_data_schema import rig
+
+from ..core import BaseEtl
+
+
+class NeuropixelsRigException(Exception):
+    """General error for MVR."""
+
+
+# assembly name, serial number, height, width, frame rate
+MVRCamera = tuple[str, str, int, int, float]
+SyncChannel = tuple[int, str]  # di line index, name
+# device name, sample rate, channels
+SyncContext = tuple[str, float, list[SyncChannel]]
+RigContext = tuple[dict, list[MVRCamera], SyncContext]
+
+
+class NeuropixelsRigEtl(BaseEtl):
+    """Neuropixels rig ETL class. Extracts information from rig-related files
+    and transforms them into an aind-data-schema rig.Rig instance.
+    """
+
+    def __init__(
+        self,
+        input_source: pathlib.Path,
+        output_directory: pathlib.Path,
+    ):
+        """Class constructor for Neuropixels rig etl class.
+
+        Parameters
+        ----------
+        input_source : Path
+          Can be a string or a Path
+        output_directory : Path
+          The directory where to save the json files.
+        """
+        super().__init__(input_source, output_directory)
+
+    def _load_resource(self, path: pathlib.Path) -> str:
```

Comment:
I don't think you need this function - `read_text` should say something to this effect if the file does not exist.